R/generate_qc_report.R

```diff
@@ -0,0 +1,44 @@
+#' Generate a QC report from a SingleCellExperiment object
+#'
+#' @param sce A SingleCellExperiment object that the report will describe
+#' @param sample_name The name of the sample
+#' @param outfile The output file path that will be created.
+#'   If the file name does not include an extension, ".html" will be added automatically.
+#'   Any directories in the path will be created as needed.
```

Comment:
Assuming that this is `rmarkdown::render()` behavior?

Reply:
It is. The directory stuff is strange, and what I was trying to solve for. `render()` creates directories iff you provide the `output_dir` argument. If you provide only an `output_file` argument, then it will fail if the path does not exist. But if you provide `output_file` without a path (just a file), it puts the output where the `.rmd` file was, which would be bad here, since that is wherever the package was installed. 

The file extension stuff I thought was neat to just leave alone, in the off chance we want to create pdfs or word docs. You can make it make a pdf by just using `"foo.pdf"` as your `output_file` even if the `.rmd` indicates it is for an html file.